lib/cosmos/tools/cmd_tlm_server/interface_thread.rb

```diff
@@ -183,15 +185,19 @@ def handle_connection_failed(connect_error)
       if @connection_failed_callback
         @connection_failed_callback.call(connect_error)
       else
-        Logger.error "#{@interface.name} Connection Failed: #{connect_error.class}:#{connect_error.message}"
+        Logger.error "#{@interface.name} Connection Failed: #{connect_error.formatted(false, false)}"
         case connect_error
-        when Errno::ECONNREFUSED, Errno::ECONNRESET, Errno::ETIMEDOUT, Errno::ENOTSOCK
+        when Errno::ECONNREFUSED, Errno::ECONNRESET, Errno::ETIMEDOUT, Errno::ENOTSOCK, Errno::EHOSTUNREACH
           # Do not write an exception file for these extremely common cases
         else
           if RuntimeError === connect_error and (connect_error.message =~ /canceled/ or connect_error.message =~ /timeout/)
             # Do not write an exception file for these extremely common cases
           else
-            Cosmos.write_exception_file(connect_error)
+            Logger.error connect_error.formatted
+            unless @connection_failed_messages.include?(connect_error.message)
+              Cosmos.write_exception_file(connect_error)
+              @connection_failed_messages << connect_error.message
+            end
```

Comment:
This prevents creating endless exception files as we try to reconnect over and over again.